python/lsst/ip/diffim/getTemplate.py

```diff
@@ -277,25 +278,33 @@ def run(self, *, coaddExposureHandles, bbox, wcs, dataIds, physical_filter):
         for tract in coaddExposureHandles:
             maskedImages, catalog, totalBox = self._makeExposureCatalog(coaddExposureHandles[tract],
                                                                         dataIds[tract])
+            warpedBox = computeWarpedBBox(catalog[0].wcs, bbox, wcs)
+            warpedBox.grow(5)  # to ensure we catch all relevant input pixels
             # Combine images from individual patches together.
-            unwarped = self._merge(maskedImages, totalBox, catalog[0].wcs)
+            unwarped, count = self._merge(maskedImages, warpedBox, catalog[0].wcs)
             # Delete `maskedImages` after combining into one large image to reduce peak memory use
             del maskedImages
-            potentialInput = self.warper.warpExposure(wcs, unwarped, destBBox=bbox)
+            if count == 0:
+                self.log.info("No valid pixels from coadd patches in tract %s; not including in output.",
+                              tract)
+                continue
+            warpedBox.clip(totalBox)
+            potentialInput = self.warper.warpExposure(wcs, unwarped.subset(warpedBox), destBBox=bbox)
 
             # Delete the single large `unwarped` image after warping to reduce peak memory use
             del unwarped
-            if not np.any(np.isfinite(potentialInput.image.array)):
-                self.log.info("No overlap from coadds in tract %s; not including in output.", tract)
+            if np.all(potentialInput.mask.array & potentialInput.mask.getPlaneBitMask("NO_DATA")):
+                self.log.info("No overlap from coadd patches in tract %s; not including in output.", tract)
                 continue
 
             catalogs.append(catalog)
-            warped[tract] = potentialInput
-            warped[tract].setWcs(wcs)
```

Comment:
Was this line unnecessary, because it's set in `_merge`?

Reply:
Yes, the merged result has the wcs when it is created.